plugins/updater/src/updater.rs

```diff
@@ -147,9 +147,17 @@ impl UpdaterBuilder {
         self
     }
 
-    pub fn endpoints(mut self, endpoints: Vec<Url>) -> Self {
+    pub fn endpoints(mut self, endpoints: Vec<Url>) -> Result<Self> {
+        if !self.config.dangerous_insecure_transport_protocol {
```

Comment:
I think we should create a validate endpoint function (or go with the old custom struct and do validation inside it) and call it here? Can be also used during config and we don't have risk of validation mismatch.

Reply:
shared the validation logic in c592c87e079b6fb3d53f3c2ccce60146b953acee